Runtime/Aplib/Integrations/Unity/Actions/TransformPathfinderAction.cs

```diff
@@ -0,0 +1,52 @@
+using Aplib.Core;
+using Aplib.Core.Belief;
+using System;
+using UnityEngine;
+
+namespace Aplib.Integrations.Unity.Actions
+{
+    public class TransformPathfinderAction<TBeliefSet> : UnityPathfinderAction<TBeliefSet>
+        where TBeliefSet : IBeliefSet
+    {
+        public TransformPathfinderAction(Func<TBeliefSet, Rigidbody> objectQuery,
+            Vector3 location,
+            float heightOffset = 0f,
+            Metadata metadata = null)
+            : this(objectQuery,
+                location: ConstantLocation(location),
+                heightOffset,
+                metadata
+            )
+        {
+        }
+
+        public TransformPathfinderAction(Func<TBeliefSet, Rigidbody> objectQuery,
+            Func<TBeliefSet, Vector3> location,
+            float heightOffset = 0f,
+            Metadata metadata = null)
+            : base(objectQuery, location, effect: PathfindingAction(objectQuery), heightOffset, metadata)
+        {
+        }
+
+        private static Action<TBeliefSet, Vector3> PathfindingAction(Func<TBeliefSet, Rigidbody> objectQuery)
+            => (beliefSet, destination) =>
+            {
+                Rigidbody rigidbody = objectQuery(beliefSet);
+
+                // Calculate the new direction
+                Vector3 direction = destination - rigidbody.position;
+                direction = new Vector3(direction.x, 0, direction.z);
+                direction.Normalize();
+
+                rigidbody.position = destination;
+
+                // If the direction is zero, don't rotate the object
+                if (direction == Vector3.zero)
+                {
+                    return;
+                }
```

Comment:
Is this old code or introduced here? Because I would not want curly brackets here

Reply:
Old code. Will pick this up if I make other changes